cpu/atxmega/include/periph_cpu.h

```diff
@@ -264,6 +264,68 @@ typedef struct {
     cpu_int_lvl_t int_lvl[TIMER_CH_MAX_NUMOF];  /**< Interrupt channels level */
 } timer_conf_t;
 
+/**
+ * @brief   Enable common SPI function
+ * @{
+ */
+#define PERIPH_SPI_NEEDS_INIT_CS
+/** @} */
+
+/**
+ * @brief   Define global value for undefined SPI device
+ * @{
+ */
+#define SPI_UNDEF               (UCHAR_MAX)
+/** @} */
+
+/**
+ * @brief Define spi_t data type to save data
+ * @{
+ */
+#define HAVE_SPI_T
+typedef uint8_t spi_t;
+/** @} */
+
+/**
+ * @brief SPI driver device mode
+ * @{
+ */
+typedef enum {
+    SPI_DRIVER_MODE_MASTER,
+    SPI_DRIVER_MODE_SLAVE,
+} spi_driver_mode_t;
+/** @} */
+
+/**
+ * @brief  SPI device configuration
+ * @{
+ */
+typedef struct {
+    SPI_t *dev;                     /**< pointer to the used SPI device */
+    pwr_reduction_t pwr;            /**< Power Management */
+    spi_driver_mode_t mode;         /**< SPI driver Mode */
```

Comment:
is this really needed? RIOT supports only master mode anyway.

Reply:
Drop-ed